test/mlir/test/gc/gpu-runner/XeGPU/f16_mlp_32x4096x4096x4096.mlir

```diff
@@ -0,0 +1,77 @@
+                    outs(%3 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+    %5 = arith.constant dense<0.000000e+00> : tensor<32x4096xf16>
+    %6 = tensor.empty() : tensor<32x4096xf16>
+    %7 = linalg.max ins(%5, %4 : tensor<32x4096xf16>, tensor<32x4096xf16>) 
+                    outs(%6 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+
+    %8 = tensor.empty() : tensor<32x4096xf16>
+    %9 = linalg.fill ins(%cst : f16) outs(%0 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+    %10 = linalg.matmul ins(%7, %arg3 : tensor<32x4096xf16>, tensor<4096x4096xf16>)
+                        outs(%9 : tensor<32x4096xf16>) -> (tensor<32x4096xf16>)
+    %11 = tensor.empty() : tensor<32x4096xf16>
+    %12 = linalg.add ins(%arg4, %10 : tensor<32x4096xf16>, tensor<32x4096xf16>) 
+                     outs(%11 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+    %13 = arith.constant dense<0.000000e+00> : tensor<32x4096xf16>
+    %14 = tensor.empty() : tensor<32x4096xf16>
+    %15 = linalg.max ins(%13, %12 : tensor<32x4096xf16>, tensor<32x4096xf16>) 
+                     outs(%14 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+
+    return %15 : tensor<32x4096xf16>
+  }
+
+  func.func @main() {
+    %cst0 = arith.constant 0.01 : f16
+    %cst1 = arith.constant 0.02 : f16
+
+    %0 = tensor.generate {
+      ^bb0(%i : index, %j : index):
+        tensor.yield %cst0 : f16
+    } : tensor<32x4096xf16>
+    %1 = tensor.generate {
+      ^bb0(%i : index, %j : index):
+        tensor.yield %cst0 : f16
+    } : tensor<4096x4096xf16>
+    %2 = tensor.generate {
+      ^bb0(%i : index, %j : index):
+        tensor.yield %cst1 : f16
+    } : tensor<32x4096xf16>
+    %3 = tensor.generate {
+      ^bb0(%i : index, %j : index):
+        tensor.yield %cst0 : f16
+    } : tensor<4096x4096xf16>
+    %4 = tensor.generate {
+      ^bb0(%i : index, %j : index):
+        tensor.yield %cst1 : f16
+    } : tensor<32x4096xf16>
+
+    %5 = func.call @linalg_mlp(%0, %1, %2, %3, %4) : (tensor<32x4096xf16>, tensor<4096x4096xf16>, tensor<32x4096xf16>, 
+                                                      tensor<4096x4096xf16>, tensor<32x4096xf16>) -> (tensor<32x4096xf16>)
+
+    %6 = tensor.extract_slice %5[0, 0][32, 1][1, 1] : tensor<32x4096xf16> to tensor<32xf16>
+
+    %cast = tensor.cast %6 : tensor<32xf16> to tensor<*xf16>
+    call @printMemrefF16(%cast) : (tensor<*xf16>) -> ()
+
+    return
+  }
+
+  func.func private @printMemrefF16(%ptr : tensor<*xf16>) attributes { llvm.emit_c_interface }
+}
+
+// CHECK: Unranked Memref base@{{(0x)?[-0-9a-fA-F]*}}
+// CHECK-SAME: rank = 1 offset = 0 sizes = [32] strides = [4096] data =
+// CHECK-NEXT: [8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344,  8.02344]
```

Comment:
Because of the crash, the temporary ref data is produced by `gc-cpu-pipeline`, because of there is no test for f16 in `gc-cpu-pipeline`, the correctness is not guaranteed, will replace ref data with accurate ones and investigate `gc-cpu-pipeline` in the future. 

Another problem I encountered is larger matmul requires k-tiling but current tiling for reduce axis will not add reduce op to compensate, this will lead to correctness issue as well, will avoid k-tiling(only use 2d tiling sizes for now).